File: JDStatusBarNotification-练习/JDStatusBarNotification-练习/ViewController.m
//
//  ViewController.m
//  JDStatusBarNotification-练习
//
//  Created by coco on 16/5/27.
//  Copyright © 2016年 XHJ. All rights reserved.
//

#import "ViewController.h"
#import <JDStatusBarNotification.h>
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
   
    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(5.0f * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
        [JDStatusBarNotification showWithStatus:@"刷新成功" dismissAfter:1.5f styleName:JDStatusBarStyleSuccess];
    });
}
- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(1.0f * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
//        [JDStatusBarNotification showWithStatus:@"刷新成功" dismissAfter:1.5f styleName:JDStatusBarStyleSuccess];
        [JDStatusBarNotification showWithStatus:@"加载错误" dismissAfter:1.5 styleName:JDStatusBarStyleError];
    });
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: dispatch_syn测试/dispatch_syn测试/ViewController.swift
//
//  ViewController.swift
//  dispatch_syn测试
//
//  Created by coco on 16/10/13.
//  Copyright © 2016年 XHJ. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        testDispatch_syn()
    }
    func testDispatch_syn() {
        DispatchQueue.global(qos: .default).async {
            print("1")
            DispatchQueue.main.sync {
                print("2")
            }
            print("3")
        }
        print("4")
        while true {
            
        }
        print("5")
    }

}

